### PR TITLE
Add singlestore upsert and existing

### PR DESCRIPTION
**Pull request by apeng-singlestore**
Adds support for the singlestore vectorstore in langchain

### CODE REVIEW

**Comment by chungyau97** [comment]
Hi @apeng-singlestore, could you share screenshots/videos of results implementing `Singlestore`?

**Comment by apeng-singlestore** [comment]
Hey, this is what it looks like!

Screenshot 2023-07-14 at 3 23 16 PM

**Comment by chungyau97** [comment]
Hi @apeng-singlestore,

could you run `yarn lint-fix` then git push again?

File: package.json [hunk 1]
```diff
@@ -48,9 +47,13 @@
...
         "rimraf": "^3.0.2",
         "run-script-os": "^1.1.6",
+        "turbo": "1.7.4",
         "typescript": "^4.8.4"
     },
     "engines": {
         "node": ">=18.15.0"
+    },
+    "dependencies": {
+        "mysql2": "^3.5.1"
```
Comment (HenryHengZJ) [inline package.json:57]: should be in `packages/components/package.json`

Comment (apeng-singlestore) [reply package.json]: Moved it!

**Review by HenryHengZJ — APPROVED** [review]
thanks @apeng-singlestore !

**Pull request merged**